### ISSUE TITLE
memory used for RHEL 7.6

### ISSUE THREAD

**Issue by battousai-dev** [oshi/oshi#1249]
Hi,

I have different result with sigar vs oshi in terms of memory used. 
oshi data:
{"Memory"
:{"Status":"OK",
"swapFree":653783040,
"memFree":5608771584,
"swapUsed":419954688,
"memTotal":21034377216,
"swapTotal":1073737728,
**"memUsed":15425605632}}**

sigar data:
{"Memory":{"Status":"OK",
"swapFree":653783040,
"memFree":403054592,
"swapUsed":419954688,
"memTotal":21034377216,
"swapTotal":1073737728,
**"memUsed":20631322624}}**

i executed the 'free' command blackhole, the result was:
                  total              used            free            shared        buff/cache      available
Mem:       20541384       9192192      338656       5408372     11010536       5448900
Swap:       1048572        410112         638460

How do you compute the memory used? Are you using the command free?

**Comment by dbwiddis** [comment]
OSHI gets its memory stats from the same source as Sigar, the `/proc/meminfo` pseudo-file.  The difference is that SIGAR, as well as the `free` command, report "free" memory, which is a mostly useless statistic, while OSHI reports "available" memory, which is actually useful and represents system usage.

I actually came to this project *because* of the fact that "Free" memory was a useless metric.  See #23, the initial issue I filed a little over 5 years ago. :)  Basically, the entire reason OSHI is what it is today is because both Java's OperatingSystemMXBean and SIGAR's "free" memory info were (and still are) both useless on Linux.

"Free" memory in Linux basically means "never used".  Once it's used, it never goes back to being "Free" even though it is "Available" to be reassigned.  Available is actually preferable from a performance standpoint!  If you operate a Linux machine for any significant length of time, you will see "Free" memory drop to near zero.  

While `/proc/meminfo` is the source OSHI uses, you can see the same stats in the `free` command line, but note the "available" column:
```
                  total              used            free            shared        buff/cache      available
Mem:       20541384       9192192      338656       5408372     11010536       5448900
```
In that output, "free" is only 330 MiB (1.6%) but "available" is 5.2 GiB (26.5%).  In reality, a lot of the "buff/cache" memory is available for use... but until a program requests it, it will happily keep some of the old "previously used" memory holding on to those pages it had.  It is "Reclaimable" memory.

Choice of this statistic is backed up by [this commit to the Linux kernel](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773) creating the "MemAvailable" entry:

> Currently, the amount of memory that is available for a new workload,
> without pushing the system into swap, can be estimated from MemFree,
> Active(file), Inactive(file), and SReclaimable, as well as the "low"
> watermarks from /proc/zoneinfo.
> 
> However, this may change in the future, and user space really should not
> be expected to know kernel internals to come up with an estimate for the
> amount of free memory.
> 
> It is more convenient to provide such an estimate in /proc/meminfo.  If
> things change in the future, we only have to change it in one place.

For more info, see [here](https://www.linuxatemyram.com/).

Bottom line:  SIGAR is wrong (well, "right" but useless).  The OperatingSystemMXBean is wrong ("right" but useless).   OSHI is providing the most useful statistic, "Available" and that's all anyone should really care about.

**Comment by battousai-dev** [comment]
> OSHI gets its memory stats from the same source as Sigar, the `/proc/meminfo` pseudo-file. The difference is that SIGAR, as well as the `free` command, report "free" memory, which is a mostly useless statistic, while OSHI reports "available" memory, which is actually useful and represents system usage.
> 
> I actually came to this project _because_ of the fact that "Free" memory was a useless metric. See #23, the initial issue I filed a little over 5 years ago. :) Basically, the entire reason OSHI is what it is today is because both Java's OperatingSystemMXBean and SIGAR's "free" memory info were (and still are) both useless on Linux.
> 
> "Free" memory in Linux basically means "never used". Once it's used, it never goes back to being "Free" even though it is "Available" to be reassigned. Available is actually preferable from a performance standpoint! If you operate a Linux machine for any significant length of time, you will see "Free" memory drop to near zero.
> 
> While `/proc/meminfo` is the source OSHI uses, you can see the same stats in the `free` command line, but note the "available" column:
> 
> ```
>      total          used         free         shared      buff/cache     available
> Mem: 20541384       9192192      338656       5408372     11010536       5448900
> ```
> 
> In that output, "free" is only 330 MiB (1.6%) but "available" is 5.2 GiB (26.5%). In reality, a lot of the "buff/cache" memory is available for use... but until a program requests it, it will happily keep some of the old "previously used" memory holding on to those pages it had. It is "Reclaimable" memory.
> 
> Choice of this statistic is backed up by [this commit to the Linux kernel](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773) creating the "MemAvailable" entry:
> 
> > Currently, the amount of memory that is available for a new workload,
> > without pushing the system into swap, can be estimated from MemFree,
> > Active(file), Inactive(file), and SReclaimable, as well as the "low"
> > watermarks from /proc/zoneinfo.
> > However, this may change in the future, and user space really should not
> > be expected to know kernel internals to come up with an estimate for the
> > amount of free memory.
> > It is more convenient to provide such an estimate in /proc/meminfo.  If
> > things change in the future, we only have to change it in one place.
> 
> For more info, see [here](https://www.linuxatemyram.com/).
> 
> Bottom line: SIGAR is wrong (well, "right" but useless). The OperatingSystemMXBean is wrong ("right" but useless). OSHI is providing the most useful statistic, "Available" and that's all anyone should really care about.

thanks for prompt replied.
